docs/source/generics.rst

```diff
@@ -370,6 +370,36 @@ type variables defined with special keyword arguments ``covariant`` or
    my_box = Box(Cat())
    look_into(my_box)  # OK, but mypy would complain here for an invariant type
 
+Note that the only change permitted in the variance of the generic class type
+variable as you move down the inheritance hierarchy is from covariant or
+contravariant to invariant. For example, this is valid:
+
+.. code-block:: python
+
+    from typing import Generic, TypeVar, Tuple, Iterable
+
+    K = TypeVar('K')
+    class CovariantSet(Generic[K_co]):
+        def items(self) -> Iterable[K_co]: ...
+
+    K_co = TypeVar('K_co', covariant=True)
+    class InvariantSet(Generic[K], CovariantSet[K]):
+        def __contains__(self, key: K) -> bool: ...
```

Comment:
Ah yes, interesting. At the same time, it uses type variable in [Enum.Flag](https://github.com/python/typeshed/blob/359c8cc313d18e309a80548c1d4643506302a525/stdlib/3.4/enum.pyi#L34), [stdlib/2/collections.pyi](https://github.com/python/typeshed/blob/359c8cc313d18e309a80548c1d4643506302a525/stdlib/2/collections.pyi#L46) and [lib-stub/typing.pyi](https://github.com/python/mypy/blob/9aef23b7ca4efe2ec4db758db61a77f3c46a6f94/test-data/unit/lib-stub/typing.pyi#L36)